tests/test_checkpoint_io/test_general_checkpoint_io.py

```diff
@@ -105,6 +110,77 @@ def test_sharded_checkpoint(use_safetensors: bool):
     recursive_check(optimizer.state_dict(), new_optimizer.state_dict())
 
 
+@pytest.mark.parametrize('use_safetensors', [True, False])
+def test_hf_load_colossalai_checkpoint(use_safetensors: bool):
+    from transformers import BertTokenizer, BertModel, BertForMaskedLM, BertConfig
+
+    ckpt_io = GeneralCheckpointIO()
+    bert_model = BertModel.from_pretrained('bert-base-chinese')
+    model_ckpt_dir = tempfile.TemporaryDirectory()
+    bert_model.config.save_pretrained(save_directory=model_ckpt_dir.name)
+    ckpt_io.save_model(bert_model, model_ckpt_dir.name, True, True, "", 10, use_safetensors=False)
+
+    new_bert_config = bert_model.config
+    new_bert_model = BertModel(config=new_bert_config)
+    new_bert_model = BertModel.from_pretrained(model_ckpt_dir.name) 
+
+    recursive_check(bert_model.state_dict(), new_bert_model.state_dict())
+
+
+@parameterize('placement_policy', ['cuda', 'cpu'])
+@parameterize('model_name', ['gpt2', 'bert'])
+@parameterize('use_safetensors', [True, False])
+def exam_state_dict(placement_policy, model_name: str, use_safetensors: bool):
+    get_components_func = non_distributed_component_funcs.get_callable(model_name)
+    model_builder, *_ = get_components_func()
+
+    with ColoInitContext(device=get_current_device()):
+        model = model_builder()
+        new_model = model_builder()
+
+    config_dict, *_ = search_chunk_configuration(model, search_range_mb=1, search_interval_byte=100)
+    chunk_manager = ChunkManager(config_dict)
+    gemini_manager = GeminiManager(placement_policy, chunk_manager)
+    model = ZeroDDP(model, gemini_manager)
+    model.train()
+
+    new_config_dict, *_ = search_chunk_configuration(new_model, search_range_mb=1, search_interval_byte=100)
+    new_chunk_manager = ChunkManager(new_config_dict)
+    new_gemini_manager = GeminiManager(placement_policy, new_chunk_manager)
+    new_model = ZeroDDP(new_model, new_gemini_manager)
+
+
+    model_ckpt_dir = tempfile.TemporaryDirectory()
+    ckpt_io = GeminiCheckpointIO()
+
+    model_size = sum(p.numel() * p.element_size() for p in model.parameters()) / 1024**2
+    ckpt_io.save_model(model, model_ckpt_dir.name, True, True, "epoch", (model_size / 3), use_safetensors=use_safetensors)
+
+    # load model
+    if ckpt_io.coordinator.is_master():
+        ckpt_io.load_model(new_model, model_ckpt_dir.name, strict=True)
+        model.to('cpu')
+        model_dict = model.state_dict(only_rank_0=True)
+        new_model.to('cpu')
+        new_model_dict = new_model.state_dict(only_rank_0=True)
+        recursive_check(model_dict, new_model_dict)
+
```

Comment:
Please check whether state dict saved by Gemini is the same as the general state dict.